.eleventy.js

```diff
@@ -1,7 +1,141 @@
+		return s
+			.trim()
+			.toLowerCase()
+			.replace(/[\s+~/^]/g, "-")
+			.replace(/[().`,%·""!?¿:@*]/g, "")
+	}
+
+	// Function for the various styles of special containers we will introduce with markdown-it-container.
+	//  This must be declared before we get into the rest of the markdown-it configuration because it is used several times.
+	//   TODO: it may be easier to contain this closer to its effect using the new `eleventyConfig.amendLibrary("md", mdLib => mdLib.enable("code"));` syntax, which arrived in 11ty v2.0.
+	const markdownItContainerCfg = (style) => {
+		return {
+			render: (tokens, idx, _options, env, slf) => {
+				if (tokens[idx].nesting === 1) {
+					tokens[idx].attrJoin("class", "callout")
+					tokens[idx].attrJoin("class", "callout-" + style)
+				}
+				return slf.renderToken(tokens, idx, _options, env, slf)
+			}
+		}
+	}
+
+	// Set up the markdown library.
+	//  This includes configuring it to use many plugins for various features (TOCs, header links, etc).
+	//   Each plugin is configured in its own `.use(...)` call -- they can be added or removed independently.
+	eleventyConfig.setLibrary("md", require("markdown-it")({
+			html: true,     // Allow html in the markdown.  Disable if your content is not trusted.
+			linkify: true,  // Autoconvert URL-like text to links.
+			replaceLink: function(link, env) {
+				// This function transforms links to `*.md` files in the output.
+				//  It allows you to link to source files as they are (and such that links work in e.g. github),
+				//  and also work in the directory structure of the output html.
+				//  Seealso: https://github.com/11ty/eleventy/issues/1204
+				const matcher = new RegExp("^(\./|\.\./|/)(.*?)(.md)(\#.*?)?$");
+				if (matcher.test(link)) {
+					// Pages with file name index.md are output at a different level in the
+					// directory hierarchy than other pages so we need to fix up relative links
+					// in those pages to their siblings so they connect.
+					const indexPage = env.page.inputPath.endsWith("/index.md");
+					const translatedLink = link.replace(matcher, "$1$2$4");
+					return indexPage ? translatedLink : translatedLink.replace(/^\.\/(.*?)/, "../$1")
+				} else {
+					return link;
+				}
+			}
+		})
+
+		// markdown-it-anchor puts page-local anchors onto headings -- a must-have.
+		.use(require("markdown-it-anchor"), {
+			permalink: true,
+			// Don't put any content in the anchor.  We'll add a visual character in CSS instead.
+			//  (This is important because the anchor tag is placed inside the hN tag.)
+			//  Consider using CSS like: `a.header-anchor:before { float:left; margin-left:-1rem; content:"🔗"; }`.
+			permalinkSymbol: "",
+			permalinkSpace: false, // Again, please don"t add actual text to the inside of the hN tag.
+			permalinkBefore: true,
+			// Generate anchors for everything but h1.  h1 tags are for page titles, and are generally not useful to jump to, so don't make anchors for those.
+			level: [2, 3, 4, 5, 6],
```

Comment:
I find it kind of confusing that headings are automatically anchor-tags, I guess it's nice if you want to copy a link which you can do by right-clicking it. But design-wise I would rather have an anchor-icon next to it that has the link which you can copy.

Reply:
You do like the anchor icon to appear even next to the h1 with the page title in it, you mean?

Reply:
Yes, that would be nice! Instead of the title appearing as a link

Reply:
But we can add that in a future PR